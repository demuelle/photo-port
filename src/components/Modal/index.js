function Modal({currentPhoto, onClose}) {

    const {name, category, description, index} = currentPhoto;

    return (
        <div className="modalBackdrop">
            <div className="modalContainer">
                <h3 className="modalTitle" data-testid="modal-title">{name}</h3>
                <img
                    src={require(`../../assets/large/${category}/${index}.jpg`).default}
                    alt={description} 
                    className="img-thumbnail mx-1"
                    key={name}
                    />
                <p>
                {description}
                </p>
                <button data-testid="modal-close-button" type="button" onClick={onClose}>
                    Close this modal
                </button>
            </div>
        </div>
    )
}

export default Modal;