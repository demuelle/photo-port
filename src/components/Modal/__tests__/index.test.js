import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Modal from '..'

const photo = { name: "picture", category: "portraits", description: "meow", index: 2 };
const onClose = jest.fn();

afterEach(cleanup)

describe('Modal is rendering', () => {

  it('renders', () => {
    render(<Modal currentPhoto={photo} onClose={onClose} />);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Modal currentPhoto={photo} onClose={onClose} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders a button', () => {
    const { getByTestId } = render(<Modal currentPhoto={photo} onClose={onClose} />)
    expect(getByTestId('modal-close-button')).toHaveTextContent("Close this modal");
  })

  it('renders a h3', () => {
    const { getByTestId } = render(<Modal currentPhoto={photo} onClose={onClose} />)
    expect(getByTestId('modal-title')).toHaveTextContent("picture");
  })

});

describe('Click Event', () => {
    it('calls onClose handler', () => {
        const { getByText } = render(<Modal onClose={onClose} currentPhoto={photo} />)
        fireEvent.click(getByText('Close this modal'));
        expect(onClose).toHaveBeenCalledTimes(1);
    })
})