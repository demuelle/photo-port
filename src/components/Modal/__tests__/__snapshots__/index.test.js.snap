// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal is rendering matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
        data-testid="modal-title"
      >
        picture
      </h3>
      <img
        alt="meow"
        class="img-thumbnail mx-1"
      />
      <p>
        meow
      </p>
      <button
        data-testid="modal-close-button"
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
